### PR TITLE
Use "openUrl" to start safari in launchAndQuit() instead of using Instruments

### PR DESCRIPTION
**Pull request by Jonahss**
This way is less hacky, and removes a dependency from the project.

I also improved the openUrl() function so that it respects async better. It used to return as soon as it was called, now it waits a little bit. (I try to tail the system logs, but unfortunately still have a hardcoded wait :( on top of that)

@imurchie for review

### CODE REVIEW

File: lib/simulator-xcode-6.js [hunk 1]
```diff
@@ -306,10 +293,10 @@ class SimulatorXcode6 {
     log.debug('Attempting to launch and quit the simulator, to create directory structure');
     log.debug(`Will launch with Safari? ${safari}`);
 
+    await this.run();
+
     if (safari) {
-      await instrumentsUtil.quickLaunch(this.udid, 'com.apple.mobilesafari');
-    } else {
-      await this.run();
+      await this.openUrl('http://www.appium.io');
```
Comment (imurchie) [inline lib/simulator-xcode-6.js]: Can we open something served by appium, so we don't depend on a working internet connection?

**Comment by imurchie** [comment]
Nice! Can we get a test or two?

**Comment by Jonahss** [comment]
Ok I'll look at tests.

File: lib/simulator-xcode-6.js (hunk 1 above)
Comment (Jonahss) [inline lib/simulator-xcode-6.js]: Hmm. Except this repo is standalone, and doesn't depend on appium.
It's ok if safari can't load the url, safari will still open and will just show "safari was not able to load the page"

Comment (imurchie) [inline lib/simulator-xcode-6.js]: Good point. Ok. Is everything ok in async-land when the url can't load?

**Comment by Jonahss** [comment]
The test I added runs launchAndQuit() while passing a `true` argument to trigger safari.

I'm not so sure any other tests I could add would be useful :(

File: lib/simulator-xcode-6.js (hunk 1 above)
Comment (Jonahss) [inline lib/simulator-xcode-6.js]: that's worth testing. ill check

Comment (Jonahss) [inline lib/simulator-xcode-6.js]: Yup. Safari just says it can't load the page, and we keep going.

**Comment by Jonahss** [comment]
@imurchie ok to merge?

**Comment by imurchie** [comment]
Ok to merge, @Jonahss

File: lib/simulator-xcode-6.js [hunk 2]
```diff
@@ -460,8 +447,40 @@ class SimulatorXcode6 {
     return [newAppPath, appPath];
   }
 
-  async openUrl () {
-    throw new Error('Command "openUrl" not supported on Xcode version 6');
+  async openUrl (url) {
+    const SAFARI_BOOTED_INDICATOR = 'MobileSafari[';
```
Comment (jlipps) [inline lib/simulator-xcode-6.js]: i tend to use UPPER_SNAKE_CASE for file-level constants, not function-level constants. not sure if we care about that as a style thing.

File: lib/simulator-xcode-6.js [hunk 3]
```diff
@@ -460,8 +447,40 @@ class SimulatorXcode6 {
...
+    const EXTRA_STARTUP_TIME = 3 * 1000;
+
+    if (await this.isRunning()) {
+      await simctl.openUrl(this.udid, url);
+      await this.tailLogsUntil(SAFARI_BOOTED_INDICATOR, SAFARI_STARTUP_TIMEOUT);
+      // So sorry, but the logs have nothing else for Safari starting.. just delay a little bit
+      log.debug(`Safari started, waiting ${EXTRA_STARTUP_TIME}ms for it to fully start`);
+      await B.delay(EXTRA_STARTUP_TIME);
+      log.debug('Done waiting for Safari');
+      return;
```
Comment (jlipps) [inline lib/simulator-xcode-6.js]: what does this return doe?

**Pull request merged**